### PR TITLE
chore: adds github action for lint, spell, link, fmt checks and resolves errors accordingly

### PR DESCRIPTION
**Pull request by dutterbutter**
# What :computer: 
- Adds github action flow for checking changes for lint, formatting, spelling, links
- Address linting, formatting, links, and spelling errors produced from checks

# Why :hand:
- The docs repo currently has no protection against changes introducing spelling mistakes, broken links etc.

# Evidence :camera:
- Running the following commands:
```
yarn lint:dead
yarn lint:fmt
yarn lint:spell
yarn lint:mdl
```

# Notes :memo:
- We should add back the pre-commit husky rules 

Closes DEVRL-138

### CODE REVIEW

**Comment by MexicanAce** [comment]
doing_the_lords_work

**Comment by dutterbutter** [comment]
@uF4No @MexicanAce updated to resolve conflicts.

File: docs/tools/compiler-toolchain/solidity.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Additional zkEVM data inserted by `zksolc`:
 - `long_version`: the full `solc` version output.
 - `zk_version`: the `zksolc` version.
 - `contract/hash`: the hash of the zkEVM bytecode.
-- `contract/factory_dependencies`: bytecode hashes of contracts created in the current contract with `CREATE`. [More details here](../../reference/architecture/contract-deployment.md#note-on-factory_deps).
+- `contract/factory_dependencies`: bytecode hashes of contracts created in the current contract with `CREATE`. [More details here](../../reference/architecture/contract-deployment.md#note-on-factory-deps).
```
Comment (MexicanAce) [inline docs/tools/compiler-toolchain/solidity.md:56]: This is actually incorrect, as the URL is supposed to have an underscore:

image

**Comment by dutterbutter** [comment]
@MexicanAce weird the links you've outlined here work correctly tho?

File: docs/dev/how-to/estimate-gas.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ To estimate gas for an L1 to L2 transaction, first gather the required values:
 
    For example, the Ethers.js library has a [`getGasPrice()`](https://docs.ethers.org/v5/api/providers/provider/#Provider-getGasPrice) method available on a `Provider` object. Your language of choice should have an equivalent.
 
-2. Call the [`zks_estimateGasL1ToL2`](../../api/api.md#zks_estimategasl1tol2) method, passing the transaction data. This returns the estimated amount of L2 gas the transaction requires; commonly called **gas limit** or similar in our code and docs.
+2. Call the [`zks_estimateGasL1ToL2`](../../api/api.md#zks-estimategasl1tol2) method, passing the transaction data. This returns the estimated amount of L2 gas the transaction requires; commonly called **gas limit** or similar in our code and docs.
```
[the remark this answers is not in the document]
Comment (dutterbutter) [reply docs/dev/how-to/estimate-gas.md:32]: I think this intentional as the links resolve correctly?

[the remark this answers is not in the document]
Comment (MexicanAce) [reply docs/dev/how-to/estimate-gas.md:32]: Ah yes, apologies, I was confusing GitHub links (looking at `.md` files directly) vs viewing from the hosted website. Closing my open conversations regarding this in the PR.

File: docs/dev/how-to/send-message-l2-l1.md [hunk 3]
```diff
@@ -74,7 +74,7 @@ function proveL2MessageInclusion(
 :::tip Parameter details
 
 - `_blockNumber`: L1 batch number in which the L2 block was included; retrievable using the `getBlock` method.
-- `_index`: Index of the L2 log in the block; returned as `id` by the [`zks_getL2ToL1LogProof`](../../api/api.md#zks_getl2tol1logproof) method.
+- `_index`: Index of the L2 log in the block; returned as `id` by the [`zks_getL2ToL1LogProof`](../../api/api.md#zks-getl2tol1logproof) method.
```
[the remark this answers is not in the document]
Comment (MexicanAce) [reply docs/dev/how-to/send-message-l2-l1.md:77]: Apologies, I was confusing GitHub links (looking at .md files directly) vs viewing from the hosted website. Closing my open conversations regarding this in the PR.

File: docs/tools/compiler-toolchain/solidity.md (hunk 1 above)
Comment (MexicanAce) [reply docs/tools/compiler-toolchain/solidity.md:56]: Apologies, I was confusing GitHub links (looking at .md files directly) vs viewing from the hosted website. Closing my open conversations regarding this in the PR.

**Comment by MexicanAce** [comment]
> @MexicanAce weird the links you've outlined here work correctly tho?

@dutterbutter Yes, looks like things are working via the website. Closed the open conversations and PR LGTM 👍

**Pull request merged**